src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NServiceBus/ReceiveMessageWrapper.cs

```diff
@@ -16,38 +13,46 @@ namespace NewRelic.Providers.Wrapper.NServiceBus
     /// </summary>
     public class ReceiveMessageWrapper : IWrapper
     {
+        private const string BrokerVendorName = "NServiceBus";
+        private const string WrapperName = "ReceiveMessageWrapper";
+
+        private const int IncomingContextIndex = 0;
+
         public bool IsTransactionRequired => false;
 
         public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
         {
-            var method = methodInfo.Method;
-            var canWrap = method.MatchesAny(assemblyName: "NServiceBus.Core", typeName: "NServiceBus.InvokeHandlersBehavior", methodName: "Invoke", parameterSignature: "NServiceBus.Pipeline.Contexts.IncomingContext,System.Action");
-            return new CanWrapResponse(canWrap);
+            return new CanWrapResponse(WrapperName.Equals(methodInfo.RequestedWrapperName));
         }
 
         public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall,
             IAgent agent, ITransaction transaction)
         {
-            var incomingContext = instrumentedMethodCall.MethodCall.MethodArguments.ExtractNotNullAs<IncomingContext>(0);
-            var logicalMessage = incomingContext.IncomingLogicalMessage;
+            var incomingContext = instrumentedMethodCall.MethodCall.MethodArguments.ExtractNotNullAs<object>(IncomingContextIndex);
+
+            var logicalMessage = NServiceBusHelpers.GetIncomingLogicalMessage(incomingContext);
+
             if (logicalMessage == null)
+            {
                 throw new NullReferenceException("logicalMessage");
+            }
 
-            var headers = logicalMessage.Headers;
+            var headers = NServiceBusHelpers.GetHeaders(logicalMessage);
 
             if (headers == null)
-                throw new NullReferenceException("headers");
+            {
+                throw new NullReferenceException("headers"); // again, not sure about this...
```

Comment:
Since the other comment tied to this one with the "again" is in a different file now (or maybe it was before in the prototype branch...either way) we might want to be clearer what we're talking about with this comment.

Reply:
The comment is an artifact from copy and paste. I feel like the logic is kinda speaking for itself about the intention so I am gonna remove the comment for now.